### ISSUE TITLE
Faithfully represent stoichiometry. 

### ISSUE THREAD

**Issue by joshkh** [MICommunity/interaction-viewer#6]
The viewer should visually represent scenarios in which proteins have a stoichiometry greater than one. For an example see PCNA homotrimer: http://www.ebi.ac.uk/intact/complex/details/EBI-1208707

**Comment by bmeldal** [comment]
Hi Josh,

Oscar just embedded the complex viewer in the complex portal dev site (http://wwwdev.ebi.ac.uk/intact/complex/). The stoichiometry has gone all weird again. Please go to our old favourite, the haemoglobin complex (EBI-9008420). You see that the manual expansion I have done has now been collapsed only leaving 3 participants with rather clumsy stoichiometry annotations.

Similarly, the PCNA (EBI-1202920) is a bag of 3 participants with stoichiometry 3 when it should just be a bag of 3 participants.

Equally, I fixed the binding features in EBI-8869931 and they have been collapsed again.

Any questions regarding versioning, please ask Oscar directly. He says he used the latest deployment.

Birgit

**Comment by colin-combe** [comment]
This is just a guess, but I see Josh added controls to switch expansion 
on and off.

I think maybe now Oscar should replace

xlv.readMIJSON(data);

with

xlv.readMIJSON(data, true);

The 'true' causes it to do the expansion. The so-called 'documentation' 
I sent out before xmas is now wrong with regards this.

Can discuss tomorrow,
cheers,
Colin


On 13/01/15 15:16, bmeldal wrote:
>
> Hi Josh,
>
> Oscar just embedded the complex viewer in the complex portal dev site 
> (http://wwwdev.ebi.ac.uk/intact/complex/). The stoichiometry has gone 
> all weird again. Please go to our old favourite, the haemoglobin 
> complex (EBI-9008420). You see that the manual expansion I have done 
> has now been collapsed only leaving 3 participants with rather clumsy 
> stoichiometry annotations.
>
> Similarly, the PCNA (EBI-1202920) is a bag of 3 participants with 
> stoichiometry 3 when it should just be a bag of 3 participants.
>
> Equally, I fixed the binding features in EBI-8869931 and they have 
> been collapsed again.
>
> Any questions regarding versioning, please ask Oscar directly. He says 
> he used the latest deployment.
>
> Birgit
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/MICommunity/interaction-viewer/issues/6#issuecomment-69759155>.
>
The University of Edinburgh is a charitable body, registered in
Scotland, with registration number SC005336.

**Comment by joshkh** [comment]
Colin is correct: I added a parameter to ReadMIJSON that determines whether
or not matrix expansion occurs and I included a toggle in the UI. The
purpose of the toggle is to give curators a before-and-after view of the
complexes. Its presence is up for debate (let's chat about it tomorrow).

Since we haven't officially given the master branch a version number I
didn't think an update would be disruptive. The build process prefixes the
app bundle with a version number so it's something I should be paying
attention to. My apologies for any headaches caused.

Let's discuss tomorrow which builds are being used for beta testing
(version labels, pretty vs. minified code, and so on).

See you all then,
Josh
